wire-ios-request-strategy/Sources/Payloads/Processing/Helpers/MLSEventProcessor.swift

```diff
@@ -144,31 +144,96 @@ public class MLSEventProcessor: MLSEventProcessing {
             return logWarn(aborting: .processingWelcome, withReason: .missingMLSService)
         }
 
+        let oneOnOneResolver = OneOnOneResolver(
+            protocolSelector: OneOnOneProtocolSelector(),
+            migrator: OneOnOneMigrator(mlsService: mlsService)
+        )
+
+        await process(
+            welcomeMessage: welcomeMessage,
+            conversationID: conversationID,
+            in: context,
+            mlsService: mlsService,
+            oneOnOneResolver: oneOnOneResolver
+        )
+    }
+
+    func process(
+        welcomeMessage: String,
+        conversationID: QualifiedID,
+        in context: NSManagedObjectContext,
+        mlsService: MLSServiceInterface,
+        oneOnOneResolver: OneOnOneResolverInterface
```

Comment:
Why are injecting these in the method? Can't they be properties of the `MLSEventProcessor`?

Reply:
In order to have these as properties of the `MLSEventProcessor`, we would need to either:
- Inject the `MLSService` from the `ZMUserSession` down into all the classes that depend on payload processors
- Make them both optional properties

Reply:
And I believe that the injection of the `MLSService` is covered by this ticket https://wearezeta.atlassian.net/browse/WPB-5827

Reply:
Ok, all good then!